wait-for-it.sh

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env bash
```

Comment:
Adding 182 lines of bash makes me feel uneasy as there is a lot of stuff here for us to check and for us to look after as direct attributions to our code base.